### DISCUSSION TITLE
saveRelationships() does not respect dehydrated state

### DISCUSSION THREAD

**Discussion by phh** [filamentphp/filament#11805] [Help]
### Package

Form builder

### Package Version

3.2.42

### How can we help you?

So, I have a select field:

```php
Forms\Components\Select::make('user_id')
    ->dehydrated(fn (?string $state): bool => filled($state))
    ->relationship('user', 'id'),
```

The idea is if the editor does not set a user I'm creating user and assigning the `user_id` in `mutateFormDataBeforeCreate()`. It works, but since it also have a relationship assigned, Filament will first create the record with the belongsTo relationship, and after in `$this->form->model($this->getRecord())->saveRelationships();` it re-saves the relationship, but setting the relation to null.

Any suggestions to solve this or should this be considered a but report instead?

**Comment by leandrocfe** [comment]
add `->dehydrated()` after `relationship`..

**Reply by phh** [reply]
Tryed, before and after (and both at the same time). Same result :/

**Reply by Tomodo531** [reply]
@leandrocfe ``$shouldCallHooksBefore`` is set to false when saving a record in [EditRecord.php](https://github.com/filamentphp/filament/blob/3.x/packages/panels/src/Resources/Pages/EditRecord.php#L143), resulting in ``saveRelationships`` being called before ``dehydrateState``. Is there an easy way around this?
